internal/mode/static/nginx/config/upstreams.go

```diff
@@ -18,6 +19,8 @@ const (
 	nginx500Server = "unix:/var/run/nginx/nginx-500-server.sock"
 	// invalidBackendRef is used as an upstream name for invalid backend references.
 	invalidBackendRef = "invalid-backend-ref"
+	// nginxConnectionClosedServer is used as a stream backend that will close the connection.
+	nginxConnectionClosedServer = "unix:/var/run/nginx/connection-closed-server.sock"
```

Comment:
I don't think we need this anymore since we are going with the map approach.

Reply:
The map approach is for when the upstream name referenced by the server does not exist right? What about the case where the upstream has no endpoints? Would we use the same approach?

Reply:
Maybe then on the Dataplane it should just not create the upstream if it has no endpoints

Reply:
We want Nginx to return a 500 status in both cases, and the only way to do that is with the map. 

> Maybe then on the Dataplane it should just not create the upstream if it has no endpoints
I think that's an OK approach for now. We may have to change it when we implement traffic splitting, though.